### PR TITLE
Spring cleaning

### PR DESCRIPTION
**Pull request by imbstack**
Probably most important to check on this one is the commit "Update mocha and fix places where pulse-publisher did not close properly". It actually changes things.

### CODE REVIEW

File: src/exchanges.js [hunk 1]
```diff
@@ -267,6 +274,12 @@ Publisher.prototype.close = async function() {
   if (this._connecting) {
     await this._connecting;
   }
+  if (this.__reconnectTimer) {
+    clearTimeout(this.__reconnectTimer);
+  }
+  if (_sleepingTimeout) {
+    clearTimeout(_sleepingTimeout);
```
Comment (jonasfj) [inline src/exchanges.js]: Is this one necessary? If not please don't clear it... I think it's only present if thing is sleeping for 12s, you are effectively aborting that with a close...

File: src/exchanges.js [hunk 2]
```diff
@@ -267,6 +274,12 @@ Publisher.prototype.close = async function() {
   if (this._connecting) {
     await this._connecting;
   }
+  if (this.__reconnectTimer) {
+    clearTimeout(this.__reconnectTimer);
```
Comment (jonasfj) [inline src/exchanges.js:278]: let's also do `this.__reconnectTimer = null` just for good measure...

File: src/exchanges.js (hunk 1 above)
Comment (jonasfj) [reply src/exchanges.js]: In fact there could be a second publisher that breaks, because you close the first publisher. This is not good.

**Review by jonasfj — CHANGES_REQUESTED** [review]
Are you sure about the  `_sleepingTimeout` part?

File: src/exchanges.js (hunk 1 above)
Comment (imbstack) [reply src/exchanges.js]: Without this the tests hang for 12 seconds before finishing.

**Review by jonasfj — APPROVED** [review]

**Pull request merged**